src/matrixdata.py

```diff
@@ -34,16 +84,55 @@ def set_pixel_value(self, row, col, val):
             self.green_matrix_data[row][cIndex] |= (1 << byteIndex)
         if val & BLUE_VALUE:
             self.blue_matrix_data[row][cIndex] |= (1 << byteIndex)
-        
-        self.dirty_bytes_array.append((row,col//8))
-    
+
+        self.dirty_bytes_array.append((row, col//8))
+
     def is_out_of_bounds(self, row, col):
+        '''
+        Check if a pixel is out of LED matrix bounds.
+
+        Parameters
+        ----------
+        row : int
+            Row value of pixel (y value).
+        col : int
+            Column value of pixel (x value).
+
+        Returns
+        -------
+        bool
+            Pixel is out of bounds?
+        '''
         return (row < 0 or row >= self.row_size or col < 0 or col >= self.col_size)
 
     def clear_dirty_bytes(self):
+        '''
+        Reset dirty pixels.
+
+        Returns
+        -------
+        None.
+        '''
         for index in self.dirty_bytes_array:
             self.red_matrix_data[index[0]][index[1]] = 0
             self.green_matrix_data[index[0]][index[1]] = 0
             self.blue_matrix_data[index[0]][index[1]] = 0
-            
+
+        self.dirty_bytes_array = []
+
+    def clear_all_bytes(self):
+        '''
+        Reset all pixels.
+
+        Returns
+        -------
+        None.
+        '''
+        self.red_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        self.green_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        self.blue_matrix_data = [bytearray(self.col_bytes) for x in range(self.row_size)]
+        # self.red_matrix_data = [bytearray(self.col_bytes)] * self.row_size
```

Comment:
Same with these commented lines?

Reply:
Done!

Reply:
I'm not sure why it wouldn't work, but I suspect it might actually produce a list of references to the same bytearray object instead of creating one bytearray object for each row. It seems like that would still run, but just set every row the same. That's just a guess though, I haven't tried it.

Reply:
This is it, you are absolutely right! If fully explains what I am seeing then. Thank you again!